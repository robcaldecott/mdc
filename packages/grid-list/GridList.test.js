import React from "react";
import {
  GridList,
  GridListTiles,
  GridTile,
  GridTilePrimary,
  GridTilePrimaryContent,
  GridTileSecondary,
  GridTileTitle,
  GridTileSupportText,
  GridTileIcon
} from ".";
import renderer from "react-test-renderer";

it("renders", () => {
  expect(
    renderer.create(
      <GridList
        tileGutter1
        headerCaption
        twoLineCaption
        icon="start"
        aspectRatio="16x9"
      >
        <GridListTiles>
          {[...Array(6).keys()].map(index => (
            <GridTile key={index}>
              <GridTilePrimary>
                <GridTilePrimaryContent>
                  <img src="image.png" alt="tile" />
                </GridTilePrimaryContent>
              </GridTilePrimary>
              <GridTileSecondary>
                <GridTileIcon>star_border</GridTileIcon>
                <GridTileTitle>
                  Tile {index + 1} with a very long title
                </GridTileTitle>
                <GridTileSupportText>
                  Support text {index + 1}
                </GridTileSupportText>
              </GridTileSecondary>
            </GridTile>
          ))}
        </GridListTiles>
      </GridList>
    )
  ).toMatchSnapshot();
});
