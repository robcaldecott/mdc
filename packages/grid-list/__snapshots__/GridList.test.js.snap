// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="mdc-grid-list mdc-grid-list--tile-gutter-1 mdc-grid-list--header-caption mdc-grid-list--twoline-caption mdc-grid-list--with-icon-align-start mdc-grid-list--tile-aspect-16x9"
>
  <ul
    className="mdc-grid-list__tiles"
  >
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          1
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          1
        </span>
      </span>
    </li>
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          2
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          2
        </span>
      </span>
    </li>
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          3
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          3
        </span>
      </span>
    </li>
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          4
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          4
        </span>
      </span>
    </li>
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          5
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          5
        </span>
      </span>
    </li>
    <li
      className="mdc-grid-tile"
    >
      <div
        className="mdc-grid-tile__primary"
      >
        <img
          alt="tile"
          className="mdc-grid-tile__primary-content"
          src="image.png"
        />
      </div>
      <span
        className="mdc-grid-tile__secondary"
      >
        <i
          className="mdc-grid-tile__icon material-icons"
        >
          star_border
        </i>
        <span
          className="mdc-grid-tile__title"
        >
          Tile 
          6
           with a very long title
        </span>
        <span
          className="mdc-grid-tile__support-text"
        >
          Support text 
          6
        </span>
      </span>
    </li>
  </ul>
</div>
`;
